src/Arcus.WebApi.Logging.Core/Correlation/HttpCorrelationClientOptions.cs

```diff
@@ -56,5 +57,21 @@ public string TransactionIdHeaderName
                 _transactionIdHeaderName = value;
             }
         }
+
+        /// <summary>
+        /// Gets the telemetry context used during HTTP dependency tracking.
+        /// </summary>
+        internal Dictionary<string, object> TelemetryContext { get; private set; } = new Dictionary<string, object>();
+
+        /// <summary>
+        /// Adds a telemetry context while tracking the HTTP dependency.
+        /// </summary>
+        /// <param name="telemetryContext">The dictionary with contextual information about the dependency telemetry.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="telemetryContext"/> is <c>null</c>.</exception>
+        public void AddTelemetryContext(Dictionary<string, object> telemetryContext)
```

Comment:
Would it be beneficial to have an overload that just takes a key/value pair, instead of a dictionary ?

Question that arises: what happens when using that overload in combination with the method that takes a complete dictionary.
IMHO, this method should not set / overwrite the `TelemetryContext` property.  Instead, it should take the key/value pairs from the `telemetryContext` parameter, and set them in the `TelemetryContext` property.  (If an entry already exists: overwrite it ?)

Reply:
Usually, when this is used, there is already a telemetry context available that is being used for other tracking as well, therefore it would be cumbersome to only have key/value pair instead of a dictionary. We could, however, when the need arises go that path.

Adding instead of overwritting, that could indeed be an improvement. 👍

Reply:
I meant to have an overload that only accepts key/value pairs next to the method that is introduced here.  But I can see that it might not be necessary